### ISSUE TITLE
[question] how to update npm for local project

### ISSUE THREAD

**Issue by unickq** [Schniz/fnm#953]
Hey,

Switching from Volta, looking for a way to pin npm version for the project.

```bash
fnm install 14
node --version > .node-version

node -v
v14.21.3

npm -v
6.14.18
```

Can the latest npm pin to the project somehow?
Running `npm install -g npm@8` from the project dir doesn't work.

**Comment by sambauers** [comment]
Use `corepack` by enabling it after installing a new Node version:

```sh
$ corepack enable
```

And in your `package.json` specify the package manager and version you want:

```json
{
  ...
  "packageManager": "npm@6.14.18",
  ...
}
```

Corepack will automatically install and use the version you specify. Not sure if Corepack is available for Node 14 though.

**Comment by mikicho** [comment]
@sambauers 
It's working for me for `pnpm`, but not for `npm`.
Seems like the symlink for `npm` is incorrect. I'm not sure if this bug is from `fnm` or `corepack` side.
![image](https://github.com/Schniz/fnm/assets/11459632/dc9f4c95-c6e7-4b7a-aea2-3034a6dcf2da)

**Comment by mikicho** [comment]
Ok. 
It seems like this is from `corepack` side: https://github.com/nodejs/corepack/issues/138

I'm wondering if we need to add `--corepack-enabled-all` option
WDYT?

**Comment by sambauers** [comment]
Right, I wasn't aware until now that corepack doesn't support npm.

Maybe setting the npm version in "engines" inside package.json will help here, and then non-corepack logic will take care of it.

**Comment by mikicho** [comment]
@sambauers it supports `npm|. Just not by default.
You need to run `corepack enable --all`

**Comment by sambauers** [comment]
It doesn't officially support it. I can't see a way to hook in to fnm to be able to run that command. It needs to be a feature request for the author I think.

**Comment by mikicho** [comment]
@Schniz WDYT about this? 
Another option is to add an install hook
